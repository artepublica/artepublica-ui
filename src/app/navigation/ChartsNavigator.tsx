import { Entypo } from '@expo/vector-icons';
import { createDrawerNavigator } from '@react-navigation/drawer';
import { DrawerActions } from '@react-navigation/native';
import { StyleSheet, TouchableOpacity } from 'react-native';

import { Text } from '@components';
import { Charts } from '@pages';
import { Theme, useTheme } from '@utils';

const DrawerNavigator = createDrawerNavigator<ChartsNavigatorParamsList>();

export type ChartsNavigatorParamsList = {
    Home: undefined;
    Chart1: undefined;
};

type ChartsNavigatorProps = {
    testOnly_initialRouteName?: keyof ChartsNavigatorParamsList;
};

export function ChartsNavigator({ testOnly_initialRouteName }: ChartsNavigatorProps): JSX.Element {
    const { theme } = useTheme();

    const style = styles(theme);

    return (
        <DrawerNavigator.Navigator
            initialRouteName={testOnly_initialRouteName ?? 'Home'}
            screenOptions={{
                drawerPosition: 'left',
                drawerActiveTintColor: theme.navigation.active,
                drawerInactiveTintColor: theme.navigation.inactive,
            }}
        >
            <DrawerNavigator.Screen
                name="Home"
                component={Charts}
                options={({ navigation }) => ({
                    headerShown: true,
                    headerTitle: () => <Text style={style.title}>Charts Home</Text>,
                    headerLeft: () => (
                        <TouchableOpacity testID="home-menu" style={{ paddingLeft: 16 }} onPress={() => navigation.dispatch(DrawerActions.openDrawer())}>
                            <Entypo name="menu" size={24} color={theme.text.textColor} />
                        </TouchableOpacity>
                    ),
                })}
            />
            <DrawerNavigator.Screen
                name="Chart1"
                component={Charts}
                options={({ navigation }) => ({
                    headerShown: true,
                    headerTitle: () => <Text style={style.title}>Chart1</Text>,
                    headerLeft: () => (
                        <TouchableOpacity testID="chart1-menu" style={{ paddingLeft: 16 }} onPress={() => navigation.dispatch(DrawerActions.openDrawer())}>
                            <Entypo name="menu" size={24} color={theme.text.textColor} />
                        </TouchableOpacity>
                    ),
                })}
            />
        </DrawerNavigator.Navigator>
    );
}

function styles(theme: Theme) {
    return StyleSheet.create({
        title: {
            fontSize: theme.header.fontSize,
            lineHeight: theme.header.lineHeight,
        },
    });
}
