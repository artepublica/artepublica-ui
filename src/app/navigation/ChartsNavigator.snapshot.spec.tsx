import { NavigationContainer } from '@react-navigation/native';
import { create } from 'react-test-renderer';

import { ChartsNavigator } from './ChartsNavigator';

describe('<ChartsNavigator />', () => {
    it('default snapshot', () => {
        const json = create(
            <NavigationContainer>
                <ChartsNavigator />
            </NavigationContainer>,
        ).toJSON();

        expect(json).toMatchSnapshot();
    });

    it('chart1 snapshot', () => {
        const json = create(
            <NavigationContainer>
                <ChartsNavigator testOnly_initialRouteName="Chart1" />
            </NavigationContainer>,
        ).toJSON();

        expect(json).toMatchSnapshot();
    });
});
