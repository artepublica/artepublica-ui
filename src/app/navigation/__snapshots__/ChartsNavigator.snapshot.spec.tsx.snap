// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartsNavigator /> chart1 snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      handlerTag={2}
      handlerType="PanGestureHandler"
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            },
          ]
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <RNSScreenNavigationContainer
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <RNSScreen
              activityState={2}
              collapsable={false}
              forwardedRef={[Function]}
              gestureResponseDistance={
                Object {
                  "bottom": -1,
                  "end": -1,
                  "start": -1,
                  "top": -1,
                }
              }
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "zIndex": 0,
                }
              }
            >
              <View
                accessibilityElementsHidden={false}
                importantForAccessibility="auto"
                style={
                  Array [
                    Object {
                      "backgroundColor": "rgb(242, 242, 242)",
                      "flex": 1,
                    },
                    Array [
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <RNCSafeAreaView
                    edges={
                      Array [
                        "bottom",
                        "left",
                        "right",
                        "top",
                      ]
                    }
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "white",
                            "fontFamily": "fontFamily",
                          },
                          undefined,
                        ]
                      }
                    >
                      Charts
                    </Text>
                  </RNCSafeAreaView>
                </View>
                <View
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    collapsable={false}
                    pointerEvents="box-none"
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "zIndex": 0,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "backgroundColor": "rgb(255, 255, 255)",
                          "borderBottomColor": "rgb(216, 216, 216)",
                          "flex": 1,
                          "shadowColor": "rgb(216, 216, 216)",
                          "shadowOffset": Object {
                            "height": 0.5,
                            "width": 0,
                          },
                          "shadowOpacity": 0.85,
                          "shadowRadius": 0,
                        }
                      }
                    />
                  </View>
                  <View
                    collapsable={false}
                    pointerEvents="box-none"
                    style={
                      Object {
                        "height": 44,
                        "maxHeight": undefined,
                        "minHeight": undefined,
                        "opacity": undefined,
                        "transform": undefined,
                      }
                    }
                  >
                    <View
                      pointerEvents="none"
                      style={
                        Object {
                          "height": 0,
                        }
                      }
                    />
                    <View
                      pointerEvents="box-none"
                      style={
                        Object {
                          "alignItems": "stretch",
                          "flex": 1,
                          "flexDirection": "row",
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        pointerEvents="box-none"
                        style={
                          Object {
                            "alignItems": "flex-start",
                            "flexBasis": 0,
                            "flexGrow": 1,
                            "justifyContent": "center",
                            "marginStart": 0,
                          }
                        }
                      >
                        <View
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                              "paddingLeft": 16,
                            }
                          }
                          testID="chart1-menu"
                        >
                          <Text />
                        </View>
                      </View>
                      <View
                        collapsable={false}
                        pointerEvents="box-none"
                        style={
                          Object {
                            "justifyContent": "center",
                            "marginHorizontal": 16,
                            "maxWidth": 590,
                          }
                        }
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "white",
                                "fontFamily": "fontFamily",
                              },
                              Object {
                                "fontSize": undefined,
                                "lineHeight": undefined,
                              },
                            ]
                          }
                        >
                          Chart1
                        </Text>
                      </View>
                      <View
                        collapsable={false}
                        pointerEvents="box-none"
                        style={
                          Object {
                            "alignItems": "flex-end",
                            "flexBasis": 0,
                            "flexGrow": 1,
                            "justifyContent": "center",
                            "marginEnd": 0,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </RNSScreen>
          </RNSScreenNavigationContainer>
        </View>
        <View
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.5)",
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {},
              Object {
                "opacity": undefined,
                "zIndex": -1,
              },
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.5)",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
              }
            }
          />
        </View>
      </View>
      <View
        accessibilityViewIsModal={false}
        onLayout={[Function]}
        removeClippedSubviews={false}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "maxWidth": "100%",
            },
            Object {
              "opacity": AnimatedValue {
                " __value": 0,
              },
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            },
            Array [
              Object {
                "bottom": 0,
                "position": "absolute",
                "top": 0,
                "width": "80%",
              },
              Object {
                "left": AnimatedValue {
                  " __value": -320,
                },
              },
              Object {
                "zIndex": 0,
              },
            ],
            Array [
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "width": 320,
              },
              false,
              undefined,
            ],
          ]
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Array [
              Object {
                "paddingEnd": 0,
                "paddingStart": 0,
                "paddingTop": 4,
              },
              undefined,
            ]
          }
          navigation={
            Object {
              "canGoBack": [Function],
              "closeDrawer": [Function],
              "dispatch": [Function],
              "emit": [Function],
              "getId": [Function],
              "getParent": [Function],
              "getState": [Function],
              "goBack": [Function],
              "isFocused": [Function],
              "jumpTo": [Function],
              "navigate": [Function],
              "openDrawer": [Function],
              "reset": [Function],
              "setParams": [Function],
              "toggleDrawer": [Function],
            }
          }
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View>
            <View
              collapsable={false}
              style={
                Array [
                  Object {
                    "marginHorizontal": 10,
                    "marginVertical": 4,
                    "overflow": "hidden",
                  },
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 4,
                  },
                  undefined,
                ]
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "selected": false,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "padding": 8,
                      },
                      Object {
                        "borderRadius": 4,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "flex": 1,
                          "marginRight": 32,
                        },
                        Object {
                          "marginLeft": 0,
                          "marginVertical": 5,
                        },
                      ]
                    }
                  >
                    <Text
                      numberOfLines={1}
                      style={
                        Array [
                          Object {
                            "color": "rgba(28, 28, 30, 0.68)",
                            "fontWeight": "500",
                          },
                          undefined,
                        ]
                      }
                    >
                      Home
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              collapsable={false}
              style={
                Array [
                  Object {
                    "marginHorizontal": 10,
                    "marginVertical": 4,
                    "overflow": "hidden",
                  },
                  Object {
                    "backgroundColor": "rgba(0, 122, 255, 0.12)",
                    "borderRadius": 4,
                  },
                  undefined,
                ]
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "selected": true,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "padding": 8,
                      },
                      Object {
                        "borderRadius": 4,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "flex": 1,
                          "marginRight": 32,
                        },
                        Object {
                          "marginLeft": 0,
                          "marginVertical": 5,
                        },
                      ]
                    }
                  >
                    <Text
                      numberOfLines={1}
                      style={
                        Array [
                          Object {
                            "color": "rgb(0, 122, 255)",
                            "fontWeight": "500",
                          },
                          undefined,
                        ]
                      }
                    >
                      Chart1
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`<ChartsNavigator /> default snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      handlerTag={1}
      handlerType="PanGestureHandler"
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            },
          ]
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <RNSScreenNavigationContainer
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <RNSScreen
              activityState={2}
              collapsable={false}
              forwardedRef={[Function]}
              gestureResponseDistance={
                Object {
                  "bottom": -1,
                  "end": -1,
                  "start": -1,
                  "top": -1,
                }
              }
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "zIndex": 0,
                }
              }
            >
              <View
                accessibilityElementsHidden={false}
                importantForAccessibility="auto"
                style={
                  Array [
                    Object {
                      "backgroundColor": "rgb(242, 242, 242)",
                      "flex": 1,
                    },
                    Array [
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <RNCSafeAreaView
                    edges={
                      Array [
                        "bottom",
                        "left",
                        "right",
                        "top",
                      ]
                    }
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "white",
                            "fontFamily": "fontFamily",
                          },
                          undefined,
                        ]
                      }
                    >
                      Charts
                    </Text>
                  </RNCSafeAreaView>
                </View>
                <View
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    collapsable={false}
                    pointerEvents="box-none"
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "zIndex": 0,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "backgroundColor": "rgb(255, 255, 255)",
                          "borderBottomColor": "rgb(216, 216, 216)",
                          "flex": 1,
                          "shadowColor": "rgb(216, 216, 216)",
                          "shadowOffset": Object {
                            "height": 0.5,
                            "width": 0,
                          },
                          "shadowOpacity": 0.85,
                          "shadowRadius": 0,
                        }
                      }
                    />
                  </View>
                  <View
                    collapsable={false}
                    pointerEvents="box-none"
                    style={
                      Object {
                        "height": 44,
                        "maxHeight": undefined,
                        "minHeight": undefined,
                        "opacity": undefined,
                        "transform": undefined,
                      }
                    }
                  >
                    <View
                      pointerEvents="none"
                      style={
                        Object {
                          "height": 0,
                        }
                      }
                    />
                    <View
                      pointerEvents="box-none"
                      style={
                        Object {
                          "alignItems": "stretch",
                          "flex": 1,
                          "flexDirection": "row",
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        pointerEvents="box-none"
                        style={
                          Object {
                            "alignItems": "flex-start",
                            "flexBasis": 0,
                            "flexGrow": 1,
                            "justifyContent": "center",
                            "marginStart": 0,
                          }
                        }
                      >
                        <View
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                              "paddingLeft": 16,
                            }
                          }
                          testID="home-menu"
                        >
                          <Text />
                        </View>
                      </View>
                      <View
                        collapsable={false}
                        pointerEvents="box-none"
                        style={
                          Object {
                            "justifyContent": "center",
                            "marginHorizontal": 16,
                            "maxWidth": 590,
                          }
                        }
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "white",
                                "fontFamily": "fontFamily",
                              },
                              Object {
                                "fontSize": undefined,
                                "lineHeight": undefined,
                              },
                            ]
                          }
                        >
                          Charts Home
                        </Text>
                      </View>
                      <View
                        collapsable={false}
                        pointerEvents="box-none"
                        style={
                          Object {
                            "alignItems": "flex-end",
                            "flexBasis": 0,
                            "flexGrow": 1,
                            "justifyContent": "center",
                            "marginEnd": 0,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </RNSScreen>
          </RNSScreenNavigationContainer>
        </View>
        <View
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.5)",
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {},
              Object {
                "opacity": undefined,
                "zIndex": -1,
              },
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.5)",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
              }
            }
          />
        </View>
      </View>
      <View
        accessibilityViewIsModal={false}
        onLayout={[Function]}
        removeClippedSubviews={false}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "maxWidth": "100%",
            },
            Object {
              "opacity": AnimatedValue {
                " __value": 0,
              },
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            },
            Array [
              Object {
                "bottom": 0,
                "position": "absolute",
                "top": 0,
                "width": "80%",
              },
              Object {
                "left": AnimatedValue {
                  " __value": -320,
                },
              },
              Object {
                "zIndex": 0,
              },
            ],
            Array [
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "width": 320,
              },
              false,
              undefined,
            ],
          ]
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Array [
              Object {
                "paddingEnd": 0,
                "paddingStart": 0,
                "paddingTop": 4,
              },
              undefined,
            ]
          }
          navigation={
            Object {
              "canGoBack": [Function],
              "closeDrawer": [Function],
              "dispatch": [Function],
              "emit": [Function],
              "getId": [Function],
              "getParent": [Function],
              "getState": [Function],
              "goBack": [Function],
              "isFocused": [Function],
              "jumpTo": [Function],
              "navigate": [Function],
              "openDrawer": [Function],
              "reset": [Function],
              "setParams": [Function],
              "toggleDrawer": [Function],
            }
          }
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View>
            <View
              collapsable={false}
              style={
                Array [
                  Object {
                    "marginHorizontal": 10,
                    "marginVertical": 4,
                    "overflow": "hidden",
                  },
                  Object {
                    "backgroundColor": "rgba(0, 122, 255, 0.12)",
                    "borderRadius": 4,
                  },
                  undefined,
                ]
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "selected": true,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "padding": 8,
                      },
                      Object {
                        "borderRadius": 4,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "flex": 1,
                          "marginRight": 32,
                        },
                        Object {
                          "marginLeft": 0,
                          "marginVertical": 5,
                        },
                      ]
                    }
                  >
                    <Text
                      numberOfLines={1}
                      style={
                        Array [
                          Object {
                            "color": "rgb(0, 122, 255)",
                            "fontWeight": "500",
                          },
                          undefined,
                        ]
                      }
                    >
                      Home
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              collapsable={false}
              style={
                Array [
                  Object {
                    "marginHorizontal": 10,
                    "marginVertical": 4,
                    "overflow": "hidden",
                  },
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 4,
                  },
                  undefined,
                ]
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "selected": false,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "padding": 8,
                      },
                      Object {
                        "borderRadius": 4,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "flex": 1,
                          "marginRight": 32,
                        },
                        Object {
                          "marginLeft": 0,
                          "marginVertical": 5,
                        },
                      ]
                    }
                  >
                    <Text
                      numberOfLines={1}
                      style={
                        Array [
                          Object {
                            "color": "rgba(28, 28, 30, 0.68)",
                            "fontWeight": "500",
                          },
                          undefined,
                        ]
                      }
                    >
                      Chart1
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
